identity-apps-core/apps/recovery-portal/src/main/webapp/password-recovery.jsp

```diff
@@ -54,11 +57,17 @@
 <%
     boolean error = IdentityManagementEndpointUtil.getBooleanValue(request.getAttribute("error"));
     String errorMsg = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("errorMsg"));
+    boolean isTenantQualifiedUsername = Boolean.parseBoolean(request.getParameter("isTenantQualifiedUsername"));
```

Comment:
We need to encode the `request.getParameter` usage, in order to overcome potential code injection attempts

Reply:
here the value is directly read into a parseBoolean function. There is no chance any code execution is possible here.